scripts/apps/search/directives/ItemList.js

```diff
@@ -668,6 +668,147 @@ export function ItemList(
                 }
             });
 
+            var MarkedDesksList = React.createClass({
```

Comment:
I think we should do it within this PR because moving 3 classes into a separate file, converting them to JSX and simply importing them here to be used as normal shouldn't take more than 30-45 minutes imo. For Angular's injectables we can send them via props. If there are any serious issues with doing this that I've omitted, please let me know and in that case we can leave it for another PR.